ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/UserManagement.scala

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api
+
+import com.daml.lf.data.Ref
+
+object UserManagement {
+
+  case class User(
+      id: String,
+      primaryParty: Option[Ref.Party],
```

Comment:
do you expect to keep this optional or is this just because you haven’t solved bootstrapping so far? My understanding was that the primary party is required.

Reply:
I believe for an `admin` user there might be no primary party. At least on VMBC, it doesn't yet make sense as there is no participant party, yet. Likewise, for a user that serves to export the whole ledger, a primary party might not make sense.

So, I'm likely going to update the commentary on the gRPC services to allow for not setting the primary party.

FYI: @bame-da

Reply:
In that setting, do you think tools like Daml Script, triggers, … that largely expect to act as a party can fail if you pass a user with no-primary party?

Reply:
I'd consider the `primary_party` a convenience mechanism that is not required for these tools to function, but makes them easier to use for the default case where a user has an on-ledger role.

For special users like one with `CanActAsAnyParty` rights (see https://github.com/digital-asset/daml/pull/11908), I'd expect that these tool should still be usable, but might require extra parameters or using lower-level functions; e.g., `submitMulti` instead of `submitUser` in Daml script.

I'm being slightly vague here, as I don't know the tools well enough yet. I defer to you and @bame-da to make the call.